### PR TITLE
add strip_num and strip_alpha modifiers

### PR DESCRIPTION
**Pull request by frederikhors**

### CODE REVIEW

File: modifiers/string.go [hunk 1]
```diff
@@ -89,6 +90,28 @@ func SnakeCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param
...
 }
 
+// NumCase removes all non-numeric characters. Example: "the price is €30,38" -> "3038". Note: The struct field will remain a string. No type conversion takes place.
+// From https://github.com/leebenson/conform
+func NumCase(_ context.Context, _ *mold.Transformer, v reflect.Value, _ string) error {
+	s, ok := v.Interface().(string)
+	if !ok {
+		return nil
+	}
+	v.SetString(regexp.MustCompile("[^0-9]").ReplaceAllLiteralString(s, ""))
```
Comment (deankarn) [inline modifiers/string.go:100]: Can we make the regexes global variables for reuse.

File: modifiers/modifiers.go [hunk 2]
```diff
@@ -16,6 +16,8 @@ func New() *mold.Transformer {
 	mod.Register("lcase", ToLower)
 	mod.Register("ucase", ToUpper)
 	mod.Register("snake", SnakeCase)
+	mod.Register("num", NumCase)
+	mod.Register("notnum", NotNumCase)
```
Comment (deankarn) [inline modifiers/modifiers.go:20]: Not in love with the tag and function names, suggestion:
- strip_alpha
- strip_num

**Review by deankarn — CHANGES_REQUESTED** [review]

File: modifiers/modifiers.go (hunk 2 above)
Comment (frederikhors) [reply modifiers/modifiers.go]: I agree. Done.

File: modifiers/string.go (hunk 1 above)
Comment (frederikhors) [reply modifiers/string.go]: I agree. Done.

**Comment by frederikhors** [comment]
I added here unicode ones.

File: modifiers/string.go [hunk 3]
```diff
@@ -99,6 +101,58 @@ func TitleCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param
...
+		return nil
+	}
+	v.SetString(stripNumRegex.ReplaceAllLiteralString(s, ""))
+	return nil
+}
+
+var stripAlphaRegex = regexp.MustCompile("[0-9]")
+
+// StripNumCase removes all numbers. Example "39472349D34a34v69e8932747" -> "Dave". Note: The struct field will remain a string. No type conversion takes place.
+// From https://github.com/leebenson/conform
```
Comment (deankarn) [inline modifiers/string.go:120]: I don't think these from statements are needed here if we can remove.

Comment (frederikhors) [reply modifiers/string.go]: Done.

File: modifiers/string.go [hunk 4]
```diff
@@ -99,6 +101,58 @@ func TitleCase(ctx context.Context, t *mold.Transformer, v reflect.Value, param
...
+func StripNumCase(_ context.Context, _ *mold.Transformer, v reflect.Value, _ string) error {
+	s, ok := v.Interface().(string)
+	if !ok {
+		return nil
+	}
+	v.SetString(stripAlphaRegex.ReplaceAllLiteralString(s, ""))
+	return nil
+}
+
+var StripNumUnicodeRegex = regexp.MustCompile(`[^\pL]`)
```
[the remark this answers is not in the document]
Comment (frederikhors) [reply modifiers/string.go]: Done.

**Pull request merged**